packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -41,20 +43,35 @@ export class PreferenceProvider implements Disposable {
         this.toDispose.dispose();
     }
 
-    protected fireOnDidPreferencesChanged(): void {
-        this.onDidPreferencesChangedEmitter.fire(undefined);
+    get<T>(preferenceName: string, resourceUri?: string): T | undefined {
+        const value = this.getPreferences(resourceUri)[preferenceName];
+        if (value !== undefined && value !== null) {
+            return value;
+        }
     }
 
-    getPreferences(): { [p: string]: any } {
-        return [];
+    getPreferences(resourceUri?: string): { [p: string]: any } {
+        return {};
     }
 
-    setPreference(key: string, value: any): Promise<void> {
+    setPreference(key: string, value: any, resourceUri?: string): Promise<void> {
         return Promise.resolve();
     }
 
     /** See `_ready`.  */
     get ready() {
         return this._ready.promise;
     }
+
+    canProvide(preferenceName: string, resourceUri?: string): { priority: number, provider: PreferenceProvider } {
+        return { priority: -1, provider: this };
```

Comment:
Why does it return itself?

Reply:
if you take a look at `FoldersPreferenceProvider` you would find this function does not always return the class itself. 

`UserPrefProvider.canProvide()` returns itself because there is only one `UserPrefProvider` per theia instance.